### PR TITLE
Effectively revert to v2.0.0

### PR DESCRIPTION
**Pull request by chadly**
This reverts most of the changes made in v3 & v4. Mainly, this gets rid of the custom test runner and relies on xunit's built-in test-running logic. This will eliminate flakiness in some of our builds.

Also removing the `ISpecification` interface and instead relying on `AsyncSpecification` & `Specification` base classes. This is necessary in order to use `IAsyncLifetime` from xunit.

### CODE REVIEW

**Comment by tullo-x86** [comment]
On the plus side, I'll be able to use ReSharper's test runner again.

File: test/TestSpecification.cs [hunk 1]
```diff
@@ -1,58 +1,48 @@
-using Xunit;
-using Xunit.Extensions;
+using System;
+using System.Threading.Tasks;
 
-[assembly: TestFramework("Xunit.Extensions.ObservationTestFramework", "Xunit.Bdd")]
-
-namespace Xunit.Bdd.Test
+namespace Xunit.Extensions.Test
```
Comment (tullo-x86) [inline test/TestSpecification.cs:4]: Why this namespace? The tests are specifically for our library, shouldn't they be in a corresponding namespace?

**Review by tullo-x86 — APPROVED** [review]

File: test/TestSpecification.cs (hunk 1 above)
Comment (chadly) [reply test/TestSpecification.cs:4]: mostly to not have to have a `using Xunit.Extensions` as the top

Comment (tullo-x86) [reply test/TestSpecification.cs:4]: That's, uh, a reason I guess... 🤷‍♂️

**Pull request merged**